### PR TITLE
Map fixes

### PR DESCRIPTION
**Pull request by atomiomi**

### CODE REVIEW

File: packages/map/src/circle/index.jsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
 
 import Layer from '../layer';
-import { useChildrenBounds } from '../map/hooks';
+import { useAddLayerBoundsToMap } from '../map/hooks';
```
Comment (suprMax) [inline packages/map/src/circle/index.jsx:5]: just `addBounds`?

File: packages/map/src/map/hooks.jsx [hunk 2]
```diff
@@ -43,6 +44,10 @@ export const useMapInstance = (nodeRef, options) => {
       isMobile: !getMedia(global, media.mediaM),
     });
 
+    if (!hasBounds) {
```
Comment (suprMax) [inline packages/map/src/map/hooks.jsx:47]: return earlier before getting map options?

File: packages/map/src/map/hooks.jsx [hunk 3]
```diff
@@ -58,47 +63,47 @@ export const useMapInstance = (nodeRef, options) => {
...
-
   useEffect(() => {
-    const boundingBox = getBoundingBox(boundsToFit);
+    const boundingBox = getBoundingBox(bounds);
 
-    if (map && boundingBox && !defaultView) {
+    if (map && boundingBox) {
       const options = { animate, padding: fitPadding };
-      if (defaultZoom) options.maxZoom = defaultZoom;
+      if (maxZoom) options.maxZoom = maxZoom;
```
Comment (suprMax) [inline packages/map/src/map/hooks.jsx:82]: do we need this option? i mean it's ok to pass it along but do we need to set it in core FE for map to work?

File: packages/map/src/map/utils.jsx [hunk 4]
```diff
@@ -44,11 +43,14 @@ export const getMapOptions = ({
 
     scrollZoom: false,
     dragRotate: false,
+    dragPan: {
+      // Disable intertia on drag
+      maxSpeed: 0,
```
Comment (suprMax) [inline packages/map/src/map/utils.jsx:48]: 👍 works

**Review by suprMax — APPROVED** [review]
LGTM overall, some minor comments

File: packages/map/CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+## v2.0.0 Breaking changes
+- Removed `Map: defaultView` prop
+- Replaced `Map: defaultZoom` prop with `Map: maxZoom` prop
```
[the remark this answers is not in the document]
Comment (peterhass) [reply packages/map/CHANGELOG.md:3]: Sorry, no idea what you mean.

You want to calculate a `maxZoom` value if neither `bounds` nor `maxZoom` is set? Makes sense to me - we shouldn't zoom out all the way by default anyway

File: packages/map/README.md [hunk 6]
```diff
@@ -41,13 +41,10 @@ const App = () => (
     {/* Credentials for maptiler */}
     credentials={{ key: 'Maptiler API secret key', map_id: 'Map style id' }}
 
-    {/* The initial center of the map */}
-    defaultView={[]}
+    {/* Max level of zoom. Can be used when there is only one marker on the map */}
```
Comment (peterhass) [inline packages/map/README.md:44]: does it really only work if only one marker is on the map?

File: packages/map/src/map/utils.jsx [hunk 7]
```diff
@@ -44,11 +43,14 @@ export const getMapOptions = ({
 
     scrollZoom: false,
     dragRotate: false,
+    dragPan: {
+      // Disable intertia on drag
```
Comment (peterhass) [inline packages/map/src/map/utils.jsx:47]: what's `intertia` ?

File: packages/map/README.md (hunk 6 above)
Comment (suprMax) [reply packages/map/README.md:44]: it's probably for setting `default` zoom/viewbox when there is only 1 marker and nothing else to show

File: packages/map/src/map/utils.jsx (hunk 7 above)
Comment (suprMax) [reply packages/map/src/map/utils.jsx:47]: by default mapbox adds a very laggy scroll inertia that does more harm than good

File: packages/map/src/circle/index.jsx (hunk 1 above)
Comment (atomiomi) [reply packages/map/src/circle/index.jsx:5]: it's hook. needs to be prefixed with use. Renamed to useAddLayerBounds

File: packages/map/CHANGELOG.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (atomiomi) [reply packages/map/CHANGELOG.md:3]: checked coreFE code. We always use zoom 14 when there is only one marker. Will use it as a fallback if there is only one point on map.

File: packages/map/src/map/utils.test.jsx [hunk 8]
```diff
@@ -71,18 +92,53 @@ describe('map/utils', () => {
 
     it('should set optinal options', () => {
```
Comment (peterhass) [inline packages/map/src/map/utils.test.jsx:93]: `optional` ? is it a typo?

File: packages/map/src/map/utils.test.jsx [hunk 9]
```diff
@@ -71,18 +92,53 @@ describe('map/utils', () => {
...
+    it('should set maxZoom level if bounds is only one point', () => {
+      const resultA = getMapOptions({
+        bounds: [[0, 0], [0, 0]],
+      });
+
+      const resultB = getMapOptions({
+        bounds: [[0, 0], [1, 1]],
+      });
+
+      assert.equal(resultA.maxZoom, 14, 'bounds is signle point');
```
Comment (peterhass) [inline packages/map/src/map/utils.test.jsx:114]:
```suggestion
      assert.equal(resultA.maxZoom, 14, 'bounds is single point');
```

File: packages/map/src/map/utils.test.jsx [hunk 10]
```diff
@@ -71,18 +92,53 @@ describe('map/utils', () => {
...
+    });
+
+    it('should set passed options', () => {
+      assert.deepEqual(getFitBoundsOptions({ animate: true, fitPadding: 3 })[1], {
+        animate: true,
+        padding: 3,
+      });
+    });
+
+    it('should set maxZoom if boundsing box is single point', () => {
```
Comment (peterhass) [inline packages/map/src/map/utils.test.jsx:131]:
```suggestion
    it('should set maxZoom if bounding box is single point', () => {
```

File: packages/map/src/map/utils.test.jsx [hunk 11]
```diff
@@ -71,18 +92,53 @@ describe('map/utils', () => {
...
+      const resultA = getFitBoundsOptions({
+        bounds: [[0, 0], [0, 0]],
+      })[1];
+
+      const resultB = getFitBoundsOptions({
+        bounds: [[0, 0], [1, 1]],
+      })[1];
+
+      assert.equal(resultA.maxZoom, 14, 'bounds is signle point');
```
Comment (peterhass) [inline packages/map/src/map/utils.test.jsx:140]:
```suggestion
      assert.equal(resultA.maxZoom, 14, 'bounds is single point');
```

**Review by peterhass — APPROVED** [review]
LGTM

I think it's great we got rid of the maxZoom prop.

File: packages/map/src/map/utils.jsx [hunk 12]
```diff
@@ -42,15 +48,32 @@ export const getMapOptions = ({
...
+    },
     pitchWithRotate: false,
+    fitBoundsOptions: { padding: fitPadding },
   };
 
-  if (defaultView) options.center = defaultView;
-  if (defaultZoom) options.zoom = defaultZoom;
-  if (isFilledArray(bounds)) options.bounds = getBoundingBox(bounds);
-  if (fitPadding) options.fitBoundsOptions = { padding: fitPadding };
+  if (isOnePoint(options.bounds)) {
```
Comment (suprMax) [inline packages/map/src/map/utils.jsx:63]: still allow to set min/max zoom. only set default maxZoom when no zoom prop was provided by the user

File: packages/map/src/map/utils.jsx [hunk 13]
```diff
@@ -42,15 +48,32 @@ export const getMapOptions = ({
...
+  }
 
   return options;
 };
+
+export const getFitBoundsOptions = ({ animate, fitPadding, bounds }) => {
+  const boundingBox = getBoundingBox(bounds);
+  const options = { animate, padding: fitPadding };
+
+  if (isOnePoint(boundingBox)) {
```
Comment (suprMax) [inline packages/map/src/map/utils.jsx:74]: same as above

**Review by suprMax — APPROVED** [review]
minZoom

File: packages/map/src/map/utils.jsx [hunk 14]
```diff
@@ -1,6 +1,8 @@
 import { LngLatBounds } from 'mapbox-gl';
 
 
+const DEFAULT_ZOOM_LEVEL = 14;
```
Comment (peterhass) [inline packages/map/src/map/utils.jsx:4]: move to `constants.jsx` ?

**Review by peterhass — APPROVED** [review]

File: packages/map/src/map/utils.jsx (hunk 14 above)
Comment (atomiomi) [reply packages/map/src/map/utils.jsx:4]: not worth for one constant. And it's used only here

File: packages/map/README.md [hunk 15]
```diff
@@ -41,13 +41,13 @@ const App = () => (
...
 
-    {/* The initial center of the map */}
-    defaultView={[]}
+    {/* Max zoom level */}
+    maxZoom={10}
 
-    {/* The initial zoom of the map */}
-    defaultZoom={15}
+    {/* Min zoom level */}
+    maxZoom={5}
```
Comment (suprMax) [inline packages/map/README.md:48]: minZoom

**Review by suprMax — APPROVED** [review]

File: packages/map/CHANGELOG.md [hunk 16]
```diff
@@ -0,0 +1,17 @@
+## v2.0.0 Breaking changes
+- Removed `Map: defaultView` prop
+- Removed `Map: defaultZoom` prop
+- Added optional `Map: maxZoom` prop
```
Comment (peterhass) [inline packages/map/CHANGELOG.md:4]:
```suggestion
- Added optional `Map: maxZoom` and `Map: minZoom` prop
```

**Review by peterhass — APPROVED** [review]

**Pull request merged**